docs/trace/extending-the-sdk/MyResourceDetector.cs

```diff
@@ -0,0 +1,34 @@
+// <copyright file="MyResourceDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.Resources;
+
+internal class MyResourceDetector : IResourceDetector
+{
+    public Resource Detect()
+    {
+        Console.WriteLine("Implement customized detection behavior for MyResourceDector:");
+        var attributes = new List<KeyValuePair<string, object>>
+        {
+            new KeyValuePair<string, object>("key", "val"),
+        };
+
+        Resource resource = new(attributes);
```

Comment:
nit: might be simplified to `return new Resource(attributes)`.

Reply:
I've done a minor refactor to the detect method.